File: apps/cli/src/paths.ts
import { resolve } from "node:path";
import { homedir } from "node:os";

export const LOCAL_DAP_DIR = ".dap";
export const USER_DAP_DIR = resolve(homedir(), ".dap");

export const DEFAULT_ENGINE_PORT = 7333;
export const DEFAULT_DASHBOARD_PORT = 7332;

export function localDapPath(cwd: string = process.cwd()): string {
  return resolve(cwd, LOCAL_DAP_DIR);
}

export function localConfigPath(cwd: string = process.cwd()): string {
  return resolve(localDapPath(cwd), "config.json");
}

export function localDbPath(cwd: string = process.cwd()): string {
  return resolve(localDapPath(cwd), "state.db");
}

export function localPidPath(cwd: string = process.cwd()): string {
  return resolve(localDapPath(cwd), "dap.pid");
}

File: apps/cli/src/dap_cli/paths.py
"""Standardowe ścieżki projektu DAP."""

from __future__ import annotations

from pathlib import Path

LOCAL_DAP_DIR_NAME = ".dap"
USER_DAP_DIR = Path.home() / ".dap"

DEFAULT_ENGINE_PORT = 7333
DEFAULT_DASHBOARD_PORT = 7332


def local_dap_dir(cwd: Path | None = None) -> Path:
    return (cwd or Path.cwd()) / LOCAL_DAP_DIR_NAME


def local_config_path(cwd: Path | None = None) -> Path:
    return local_dap_dir(cwd) / "config.json"


def local_db_path(cwd: Path | None = None) -> Path:
    return local_dap_dir(cwd) / "state.db"


def local_pid_path(cwd: Path | None = None) -> Path:
    return local_dap_dir(cwd) / "dap.pid"
